src/main/java/Duke.java

```diff
@@ -1,10 +1,58 @@
+import duke.command.Command;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import duke.data.Storage;
+import duke.exception.DukeException;
+import duke.object.Task;
+import duke.object.TaskList;
+import duke.user.Ui;
+import duke.util.Parser;
+
+/**
+ * A simple chatbot which provides basic to-do-list functions
+ *
+ * @author  Wallace Lim
+ * @version 0.1
+ */
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Sole constructor, loads previously stored tasks if available
+     *
+     * @param filePath storage path of previous tasks
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            this.tasks = storage.loadTasks();
+        } catch (FileNotFoundException e) {
+            System.out.println(e);
+        }
+    }
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.lnBreak(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+            } catch (DukeException | IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("C:\\Users\\Admin\\OneDrive - National University of Singapore\\School\\Y2S1\\CS2103T\\duke\\src\\main\\java\\data\\tasks.txt").run();
```

Comment:
I would recommend using a relative path for the save file path. Absolute path like this means it won't work outside of your computer :sweat_smile: